File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/BaseDto.ts
// src/models/entity/BaseDto.ts
import { z } from 'zod';
import { GuidSchema } from '../Guid';

export const BaseDtoSchema = z.object({
    id: GuidSchema,
});

export type BaseDto = z.infer<typeof BaseDtoSchema>;

export const BaseEntityWithChatsDtoSchema = BaseDtoSchema.extend({
    chatIds: z.array(GuidSchema).optional(),
});

export type BaseEntityWithChatsDto = z.infer<typeof BaseEntityWithChatsDtoSchema>;

File: src/CorporateMessengerWithAgile/Application/Dto/BaseDto.cs
using Domain.Entity;

namespace Application.Dto
{
    public abstract class BaseDto
    {
        public Guid Id { get; set; }
    }

    public abstract class BaseEntityWithChatsDto : BaseDto
    {
        public IReadOnlyList<Guid> ChatIds { get; set; } = null!;
    }
}
